torch/csrc/api/src/data/samplers/random.cpp

```diff
@@ -12,10 +12,11 @@ namespace samplers {
 RandomSampler::RandomSampler(int64_t size, Dtype index_dtype)
     : indices_(torch::randperm(size, index_dtype)) {}
 
-void RandomSampler::reset() {
+void RandomSampler::reset(optional<size_t> new_size) {
   // This allocates a new chunk of memory every time (just FYI). It should be
   // amortized over the entire epoch hopefully.
-  indices_ = torch::randperm(indices_.numel(), indices_.options());
+  indices_ = torch::randperm(
```

Comment:
Please use `new_size.value_or(indices_.numel())`. I think you might have to do a `static_cast<size_t>(indices_.numel())` in there, please check

Reply:
nit: Let's also maybe put the whole expression to get the new size on a separate line above, just to make it more readable than a long inline call